### PR TITLE
Login back-end

### PR DESCRIPTION
**Pull request by ameentalahmeh**
- Create a `login` controller which handling the `login` process:
    - [x] Firstly,  checking if the user exists or not in the database.
    - [x] Secondly, if the user exists, then compare the input password with the stored password in the database.
    - [x] Finally, If user logged, then `set a cookies`, to enable a user login without entering his password. 

- [x] Create `/login` route that sends a request to the server to enable the user to log in.

### CODE REVIEW

File: src/controllers/Auth.Student.controller/login.controller.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
+const bcrypt = require('bcryptjs');
+const { setCookies } = require('./setCookiesController');
+
+const Student = require('../../models/Student.model.js');
+
+
+// Sign in
```
Comment (RazanTayyem) [inline src/controllers/Auth.Student.controller/login.controller.js]: why you wrote this comment

**Review by RazanTayyem — APPROVED** [review]
good, I loved how to separate the codes

File: src/controllers/Auth.Student.controller/login.controller.js (hunk 1 above)
Comment (ameentalahmeh) [reply src/controllers/Auth.Student.controller/login.controller.js]: Thnx, @RazanTayyem :smile: 
It for me to mark it, forget to delete it.

File: src/controllers/Auth.controller.js [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    })
+    .catch((compareErr) => {
+      reject(compareErr);
+    });
+});
+
+const signup = (req, res) => {
+  const student = req.body;
+  if (!student) {
+    res.json({ success: false, err: 'No student data found!' });
```
Comment (ghassanmas) [inline src/controllers/Auth.controller.js]: Consdier the scinario where the `req.body` is empty, the user haven't filled any field, `req.body = {}` and hence `{}` always evalutes as `true`. 
Also the students should have multiple varibale such as gender, age..etc, so if you are trying to do server-side validation you should check for each variable

**Review by ghassanmas — APPROVED** [review]

**Pull request merged**